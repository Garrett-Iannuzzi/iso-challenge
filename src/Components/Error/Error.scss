.section-error {
  border: solid 4px black;
  background-color:rgb(225, 207, 228);
  height: 15em;
  margin: 0 auto;
  margin-top: 30px;
  text-align: center;
  width: 50%;
}

h3 {
  color:red;
  font-size: 4em;
  font-weight: bolder;
  margin: 10px 10px;
}

.p-error {
  font-size: 1em;
  margin: 10px 50px;
}

.button-error {
  font-size: .8em;
  width: 35%;
  font-weight: bolder;
  height: 35px;
  border-radius: 5px;
  margin-top: 30px;
  &:hover {
    cursor: pointer;
    border: solid 2px black;
    background-color:white;
    color: black;
  }
}