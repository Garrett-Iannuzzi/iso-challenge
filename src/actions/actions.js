export const getTeamsInfo = (teamOne, teamTwo) => ({
  type: 'GET_TEAMS',
  teamOne,
  teamTwo
});

export const getPlayerInfo = (playerInfo) => ({
  type: 'GET_PLAYERS',
  playerInfo
});

export const getStatsInfo = (stats) => ({
  type: 'GET_STATS',
  stats
});