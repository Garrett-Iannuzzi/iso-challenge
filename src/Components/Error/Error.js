import React from 'react';
import './Error.scss';
import { connect } from 'react-redux';
import { isLoggedIn } from '../../actions/actions';


export const Error = (props) => {
  const goHome = () => {
    props.isLoggedIn(null)
    return props.history.push('/')
  }
    
  return (
    <section className='section-error'>
      <div className='p-error-container'>
        <h3 className='header-error'>Error!!</h3>
        <p className='p-error'>*All fields must be filled out to login*</p>
      </div>
      <button className='button-error' onClick={ () => goHome() }>Back To Home</button>
    </section>
  )
}

export const mapDispatchToProps = dispatch => ({
  isLoggedIn: (status) => dispatch(isLoggedIn(status))
})

export default connect(null, mapDispatchToProps)(Error)
