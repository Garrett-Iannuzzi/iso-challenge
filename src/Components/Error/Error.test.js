import React from 'react';
import { Error, mapDispatchToProps } from './Error';
import { shallow } from 'enzyme';
import { isLoggedIn } from '../../actions/actions';

describe('Error', () => {

  let wrapper

  describe('Error Component', () => {

    beforeEach(() => {
      wrapper = shallow(<Error />)
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('mapDispatchToProps', () => {

    it('Should call mapDispatchToProps with correct team info when goHome is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = isLoggedIn(null);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.isLoggedIn(null);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

});

