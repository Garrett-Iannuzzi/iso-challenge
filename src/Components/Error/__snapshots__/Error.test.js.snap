// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Error Component should match the snapshot 1`] = `
<section
  className="section-error"
>
  <div
    className="p-error-container"
  >
    <h3
      className="header-error"
    >
      Error!!
    </h3>
    <p
      className="p-error"
    >
      *All fields must be filled out to login*
    </p>
  </div>
  <button
    className="button-error"
    onClick={[Function]}
  >
    Back To Home
  </button>
</section>
`;
